### PR TITLE
Clean up postgres source replication slot during end-migration and cutover workflow

### PR DESCRIPTION
**Pull request by sanyamsinghal**

### CODE REVIEW

**Comment by sanyamsinghal** [comment]
https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline/1647/

File: yb-voyager/src/srcdb/postgres.go [hunk 1]
```diff
@@ -537,3 +534,23 @@ func (pg *PostgreSQL) CreateLogicalReplicationSlot(conn *pgconn.PgConn, migratio
...
+			utils.ErrExit("failed to create replication connection for dropping replication slot: %s", err)
+		}
+		defer conn.Close(context.Background())
+	}
+	log.Infof("dropping replication slot: %s", replicationSlotName)
+	err = pglogrepl.DropReplicationSlot(context.Background(), conn, replicationSlotName, pglogrepl.DropReplicationSlotOptions{})
+	if err != nil {
+		// ignore "does not exist" error while dropping replication slot
+		if !strings.Contains(err.Error(), "does not exist") {
+			return fmt.Errorf("delete existing replication slot: %v", err)
```
Comment (amit-yb) [inline yb-voyager/src/srcdb/postgres.go:552]: Include name of the slot in the message.

File: yb-voyager/cmd/exportData.go [hunk 2]
```diff
@@ -211,6 +211,13 @@ func exportData() bool {
 					utils.ErrExit("failed to delete stream id after data export: %v", err)
 				}
 			}
+			if exporterRole == SOURCE_DB_EXPORTER_ROLE {
```
Comment (amit-yb) [inline yb-voyager/cmd/exportData.go:214]: This comment is for line 204: the `markCutoverProcessed()` should be called only after cleaning the YB streaming ID and pg replication slot. If voyager exits *after* marking cutover processed and *before* deleting the replication objects, they will be left uncleaned.

File: yb-voyager/cmd/exportData.go [hunk 3]
```diff
@@ -211,6 +211,13 @@ func exportData() bool {
...
+				msr, err := metaDB.GetMigrationStatusRecord()
+				if err != nil {
+					utils.ErrExit("get migration status record: %v", err)
+				}
+				deletePGReplicationSlot(msr, &source)
```
Comment (amit-yb) [inline yb-voyager/cmd/exportData.go:219]: What about the `PUBLICATION` objects?

**Review by amit-yb — COMMENTED** [review]
Just a couple of minor comments.

**Comment by amit-yb** [comment]
Update the PR subject line to mention that the slot is also cleaned during cutover processing.

File: yb-voyager/cmd/exportData.go (hunk 3 above)
Comment (sanyamsinghal) [reply yb-voyager/cmd/exportData.go:219]: will take care of it in a separate PR once @makalaaneesh's corresponding changes are in.

Comment (amit-yb) [reply yb-voyager/cmd/exportData.go:215]: File a ticket for the pending work.

**Review by amit-yb — APPROVED** [review]
LGTM.

**Pull request merged**